### PR TITLE
PDCL-7841 - show hidden containers on request failure

### PR DESCRIPTION
**Pull request by carterworks**
## Related Issue

[Jira ticket PDCL-7841](https://jira.corp.adobe.com/browse/PDCL-7841)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have made any necessary test changes and all tests pass.
- [x] I have run the Sandbox successfully.

### CODE REVIEW

File: test/unit/specs/components/Personalization/createComponent.spec.js [hunk 1]
```diff
@@ -136,9 +135,29 @@ describe("Personalization", () => {
     expect(viewCache.storeViews).not.toHaveBeenCalled();
   });
   it("should trigger onClickHandler at onClick", () => {
-    build();
     personalizationComponent.lifecycle.onClick({ event });
 
     expect(onClickHandler).toHaveBeenCalled();
   });
+  it("should call showContainers() when a request fails", () => {
+    const useCallbackAggregator = () => {
```
Comment (carterworks) [inline test/unit/specs/components/Personalization/createComponent.spec.js:143]: I know that this is almost exactly the same as [`createCallbackAggregator.js`](https://github.com/adobe/alloy/blob/b75317552df1deeef5e5f5f113e3b3ecdc37b70b/src/utils/createCallbackAggregator.js#L19), but I didn't want to have production code as a dependency in a test. I'm happy to change it or to move this into the [test helpers folder](https://github.com/adobe/alloy/blob/b75317552df1deeef5e5f5f113e3b3ecdc37b70b/test/unit/helpers) if that's a better place for it.

Comment (jonsnyder) [reply test/unit/specs/components/Personalization/createComponent.spec.js:143]: Can you change this to use mocks (jasmine.createSpy)? You can use `calls.argsFor` to get the function that was passed to onRequestFailure.

**Review by jonsnyder — CHANGES_REQUESTED** [review]

File: src/components/Personalization/createFetchDataHandler.js [hunk 2]
```diff
@@ -36,7 +30,6 @@ export default ({
     );
     onRequestFailure(() => {
```
Comment (ninaceban) [inline src/components/Personalization/createFetchDataHandler.js:31]: you can move this to `createComponent` in the `if (personalizationDetails.shouldFetchData())` block and not pass the `onRequestFailure` here at all.

**Review by jonsnyder — APPROVED** [review]

File: src/components/Personalization/createFetchDataHandler.js (hunk 2 above)
Comment (carterworks) [reply src/components/Personalization/createFetchDataHandler.js:31]: Good idea. I'll make that change.

**Review by ninaceban — APPROVED** [review]
Looks great! could you please another test that would assert that when `requestFails` but fetchDataHandler is triggered both the promise defer is rejected and `showContainers ` is triggered.

**Review by jonsnyder — APPROVED** [review]

**Pull request merged**